pyron-engine/src/main/scala/com/cloudentity/pyron/rule/RulesConfReader.scala

```diff
@@ -120,58 +120,69 @@ object RulesConfReader {
     }
 
   // returns list of Rules on success or non-empty list of missing fields
-  def composeRuleConfs(serviceConfs: List[ServiceRulesConf], addresses: Map[PluginName, PluginAddressPrefix]): ValidationNel[String, List[RuleConfWithPlugins]] = {
-    val validations: List[ValidationNel[String, RuleConfWithPlugins]] =
-      for {
-        (serviceConf, i)  <- serviceConfs.zipWithIndex
-        (endpointConf, j) <- serviceConf.endpoints.zipWithIndex
-      } yield {
-        composeRuleConf(addresses, ServiceConf(serviceConf.default.getOrElse(emptyRuleRawConf), serviceConf.request, serviceConf.response), endpointConf)
-          .leftMap(errorMsgs => errorMsgs.map(errorMsg => s"Service $i, endpoint $j,${endpointConf.rule.endpointName.map(n => s" '$n'").getOrElse("")} $errorMsg"))
-      }
-
-    validations.sequenceU // IntelliJ has problems with understanding sequenceU - it actually compiles
+  def composeRuleConfs(serviceConfs: List[ServiceRulesConf],
+                       addresses: Map[PluginName, PluginAddressPrefix]
+                      ): ValidationNel[String, List[RuleConfWithPlugins]] = {
+
+    val validations = for {
+      (serviceConf, i) <- serviceConfs.zipWithIndex
+      (endpointConf, j) <- serviceConf.endpoints.zipWithIndex
+    } yield {
+      val defaultConf = ServiceConf(serviceConf.default.getOrElse(emptyRuleRawConf), serviceConf.request, serviceConf.response)
+      composeRuleConf(addresses, defaultConf, endpointConf)
+        .leftMap(_.map(errorMsg => {
+          val nameStr = endpointConf.rule.endpointName.map(n => s" '$n'").getOrElse("")
+          s"Service $i, endpoint $j,$nameStr $errorMsg"
+        }))
+    }
+    validations.sequenceU
   }
 
-  def composeRuleConf(addresses: Map[PluginName, PluginAddressPrefix], defaultConf: ServiceConf, endpointConf: EndpointConf): ValidationNel[String, RuleConfWithPlugins] = {
+  def composeRuleConf(addresses: Map[PluginName, PluginAddressPrefix],
+                      defaultConf: ServiceConf,
+                      endpointConf: EndpointConf
+                     ): ValidationNel[String, RuleConfWithPlugins] = {
+
     val requestPluginsConf = composeRequestPluginsConf(addresses, defaultConf, endpointConf)
     val responsePluginConfs = composeResponsePluginsConf(addresses, defaultConf, endpointConf)
-    val endpointName = endpointConf.rule.endpointName.orElse(defaultConf.rule.endpointName)
+    composeAndValidateRequiredFields(defaultConf.rule, endpointConf.rule).map { rf =>
+      val ruleConf = getRuleConf(defaultConf, endpointConf, rf)
+      RuleConfWithPlugins(ruleConf, requestPluginsConf, responsePluginConfs)
+    }
+  }
+
+  // in the end used for validation only
+  private def getRuleConf(defaultConf: ServiceConf, endpointConf: EndpointConf, rf: RuleRequiredFields): RuleConf = {
```

Comment:
Why this method is named `getRuleConf`? What do you mean by "in the end used for validation only"? Do we need to change RulesConfReader at all? I don't see the point.

Reply:
ok, it's some stray comment I wrote when getting through this code. I will remove it. I think other changes should stay. One or two condition checks were replaced with simpler equivalent. getRuleConf was long and doing several steps, which could be in separate shorter methods. I try to keep functions short, ideally 10 lines or less. Scala can do a lot in one line even. If it doesn't fit on one screen, I forget what it's about half way through. It helped me understand what it's doing, by splitting it in smaller, more focused parts, and it's generally recommended to do so. It's now half it's initial size and delegating to named subroutine. Also, if call takes a lot of parameters, it's easier to have them vertically. Otherwise I'd need to scroll horizontally a lot. I even have a vertical line in IntelliJ to remind me of not crossing some sane line length. I agree there isn't any 'essential' change but sometimes it's essential to make it more accessible for the new reader.

Reply:
Fair enough, I guess it wouldn't be an issue if part of smaller PR ;)